perun-core/src/main/java/cz/metacentrum/perun/core/impl/DatabaseManagerImpl.java

```diff
@@ -73,4 +80,119 @@ public String getDatabaseInformation(PerunSession sess) throws InternalErrorExce
 			throw new InternalErrorException(ex);
 		}
 	}
+
+	public String getCodeDatabaseVersion(String fileName) throws InternalErrorException {
+		try(BufferedReader br = new BufferedReader(new InputStreamReader(ClassLoader.getSystemResourceAsStream(fileName)))) {
+
+			Pattern versionPattern = Pattern.compile("^[1-9][0-9]*[.][0-9]+[.][0-9]+");
+			Pattern commentPattern = Pattern.compile("^--.*");
+
+			String line = br.readLine();
+
+			while (line != null){
+				line = line.trim();
+
+				// ignore empty lines and comments at the start of the file and between versions
+				if(line.isEmpty() || commentPattern.matcher(line).matches()) {
+					line = br.readLine();
+					continue;
+				}
+				if(versionPattern.matcher(line).matches()){
+					return line;
+				} else {
+					throw new InternalErrorException("Version does not match the pattern required in " + fileName);
+				}
+			}
+			throw new InternalErrorException(fileName + " is empty");
+
+		} catch(IOException e) {
+			throw new InternalErrorException("Error reading " + fileName, e);
+		}
+	}
+
+	public void updateDatabaseVersion(String currentDBVersion, String fileName) throws InternalErrorException {
+		List<DBVersion> versions = getChangelogVersions(currentDBVersion, fileName);
+		Collections.reverse(versions);
+
+		for (DBVersion v : versions) {
+			List<String> successfulCommands = new ArrayList<>();
+			for (String c : v.getCommands()) {
+				try {
+					jdbc.execute(c);
+					successfulCommands.add(c);
+				} catch(EmptyResultDataAccessException ex) {
+					log.error("Update unsuccessful. All versions before " + v.getVersion() + " were successfully executed. " +
+							"Error executing command in version " + v.getVersion() + ": " + c, ex);
+					log.error("Successful commands from " + v.getVersion() + ": " + successfulCommands);
+					throw new ConsistencyErrorException("Update unsuccessful. Error executing command in version " + v.getVersion() + ": " + c, ex);
+				} catch(RuntimeException ex) {
+					log.error("Update unsuccessful. All versions before " + v.getVersion() + " were successfully executed. " +
+							"Error executing command in version " + v.getVersion() + ": " + c, ex);
+					log.error("Successful commands from " + v.getVersion() + ": " + successfulCommands);
+					throw new InternalErrorException("Update unsuccessful. Error executing command in version " + v.getVersion() + ": " + c, ex);
+				}
+			}
+		}
+	}
+
+	public List<DBVersion> getChangelogVersions(String currentDBVersion, String fileName) throws InternalErrorException {
+
+		Pattern versionPattern = Pattern.compile("^[1-9][0-9]*[.][0-9]+[.][0-9]+");
+		Pattern commentPattern = Pattern.compile("^--.*");
+
+		List<DBVersion> versions = new ArrayList<>();
+		boolean versionFound = false;
+
+		try(BufferedReader br = new BufferedReader(new InputStreamReader(ClassLoader.getSystemResourceAsStream(fileName)))) {
```

Comment:
There is the try-with-resources statement used, it should ensure that the resource (or stream) is closed properly.